lib/streamlit/server/routes.py

```diff
@@ -132,6 +132,11 @@ def get(self):
         if self._callback():
             self.write("ok")
             self.set_status(200)
+
+            # Manually set the _xsrf token when rechecking health.
+            if config.get_option("server.enableCSRF"):
```

Comment:
Done! 

And to close the loop, I ended up not going with the `SessionInfo` approach discussed. It would have required passing a token in `SessionInfo` and then saving the token as a cookie.

Reply:
Thanks for adding the comment!